node/sc/bridge_manager.go

```diff
@@ -372,21 +358,34 @@ func (bi *BridgeInfo) UpdateHandledNonce(nonce uint64) {
 	}
 }
 
-// AddRequestValueTransferEvents adds events into the pendingRequestEvent.
-func (bi *BridgeInfo) AddRequestValueTransferEvents(evs []*RequestValueTransferEvent) {
-	if bi.pendingRequestEvent.Len() > maxPendingNonceDiff {
-		logger.Trace("adding request value transfer events is ignored", "len", bi.pendingRequestEvent.Len(), "limit", maxPendingNonceDiff)
-		return
+// UpdateLowerHandledNonce updates the lower handle nonce.
+func (bi *BridgeInfo) UpdateLowerHandleNonce(nonce uint64) {
+	if bi.lowerHandleNonce < nonce {
+		vtLowerHandleNonceCount.Inc(int64(nonce - bi.lowerHandleNonce))
+		bi.lowerHandleNonce = nonce
 	}
+}
 
+// AddRequestValueTransferEvents adds events into the pendingRequestEvent.
+func (bi *BridgeInfo) AddRequestValueTransferEvents(evs []*RequestValueTransferEvent) {
 	for _, ev := range evs {
+		if bi.pendingRequestEvent.Len() > maxPendingNonceDiff {
+			flatten := bi.pendingRequestEvent.Flatten()
```

Comment:
Is the `Flatten()` should be done inside the `for-loop`?

Reply:
Yes it is. Because after this new event will be added. And then the max value can be changed.

I also this loop is too costly. So I will refine this later.